src/Authentication/Common.Authentication/Factories/ClientFactory.cs

```diff
@@ -246,7 +249,7 @@ public void RemoveAction(Type actionType)
             }
         }
 
-        public void AddHandler(DelegatingHandler handler)
+        public void AddHandler<T>(T handler) where T: DelegatingHandler, ICloneable
         {
             if (handler != null)
```

Comment:
Change the handlers collection to be OrderedDictionary to respect the pipeline adding order.